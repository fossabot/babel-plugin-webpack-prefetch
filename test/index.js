const { join } = require('path')
const { readdirSync, statSync, readFileSync } = require('fs')
const { transform: transformBabel7 } = require('@babel/core')
const { transform: transformBabel6 } = require('babel-core')

const FIXTURE_PATH = join(__dirname, 'fixtures')
const PLUGIN_FILE = join(__dirname, '../index.js')

const plugin = require(PLUGIN_FILE)

const testFolders = readdirSync(FIXTURE_PATH).filter(file =>
  statSync(join(FIXTURE_PATH, file)).isDirectory(),
)

function transformInput(input, config, transform) {
  const result = transform(input, {
    plugins: [[plugin, config]],
  })
  return result.code
}

describe('babel-plugin-webpack-prefetch', () => {
  testFolders.forEach(folderName => {
    let config
    try {
      config = require(join(FIXTURE_PATH, folderName, 'config.js'))
    } catch (e) {
      // don't care
    }
    const input = readFileSync(
      join(FIXTURE_PATH, folderName, 'input.js'),
      'utf8',
    )
    it(`${folderName} matches snapshot for babel 7`, () => {
      const result = transformInput(input, config, transformBabel7)
      expect(result.trim()).toMatchSnapshot(folderName)
    })
    it(`${folderName} matches snapshot for babel 6`, () => {
      const result = transformInput(input, config, transformBabel6)
      expect(result.trim()).toMatchSnapshot(folderName)
    })
  })
})
