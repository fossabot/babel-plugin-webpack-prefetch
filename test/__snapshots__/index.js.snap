// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-webpack-prefetch basic matches snapshot for babel 6: basic 1`] = `
"import ccc from 'ddd';
import 'aaa';

const a = import( /* webpackPrefetch: true */'aaa');
const b = import( /* webpackPrefetch: true */'bbb');"
`;

exports[`babel-plugin-webpack-prefetch basic matches snapshot for babel 7: basic 1`] = `
"import ccc from 'ddd';
import 'aaa';
const a = import(
/* webpackPrefetch: true */
'aaa');
const b = import(
/* webpackPrefetch: true */
'bbb');"
`;

exports[`babel-plugin-webpack-prefetch preserve-comment matches snapshot for babel 6: preserve-comment 1`] = `
"import ccc from 'ddd';
import 'aaa';

const a = import( /* webpackPrefetch: false */'aaa');
const b = import( /* webpackPrefetch: true fdfasdfdasf */'bbb');
const ddd = import( /* webpackPrefetch: true */'wow');"
`;

exports[`babel-plugin-webpack-prefetch preserve-comment matches snapshot for babel 7: preserve-comment 1`] = `
"import ccc from 'ddd';
import 'aaa';
const a = import(
/* webpackPrefetch: false */
'aaa');
const b = import(
/* webpackPrefetch: true fdfasdfdasf */
'bbb');
const ddd = import(
/* webpackPrefetch: true */
'wow');"
`;
